docs/classes.md

```diff
@@ -0,0 +1,263 @@
+        self.param = param1
+
+    def Method2(self : 'MyClass1'):
+        return MyClass(1)
+
+if __name__ == '__main__':
+    obj = MyClass1()
+    obj.Method1(obj.Method2())
+    print(obj.param.param1)
+```
+### - PYTHON _OUTPUT_
+
+```Shell
+MyClass1 Object created!
+MyClass Object created!
+1
+```
+
+### - C Header File Equivalent
+
+```C
+#ifndef SBOOF_H
+#define SBOOF_H
+
+struct MyClass {
+    int64_t param1;
+    bool is_freed;
+};
+struct MyClass1 {
+    bool is_freed;
+    struct MyClass param;
+};
+
+void MyClass__init__(struct MyClass* self, int64_t param1);
+void MyClass__del__(struct MyClass* self);
+void MyClass1__init__(struct MyClass1* self);
+void MyClass1__Method1(struct MyClass1* self, struct MyClass* param1);
+struct MyClass MyClass1__Method2(struct MyClass1* self);
+void MyClass1__del__(struct MyClass1* self);
+
+#endif // sboof_H
+```
+### - C File Equivalent
+
+```C
+/*........................................*/
+void MyClass__init__(struct MyClass* self, int64_t param1)
+{
+    self->is_freed = 0;
+    self->param1 = param1;
+    printf("%s\n", "MyClass Object created!");
+}
+/*........................................*/
+/*........................................*/
+void MyClass__del__(struct MyClass* self)
+{
+    if (!self->is_freed)
+    {
+        // pass
+        self->is_freed = 1;
+    }
+}
+/*........................................*/
+/*........................................*/
+void MyClass1__init__(struct MyClass1* self)
+{
+    self->is_freed = 0;
+    printf("%s\n", "MyClass1 Object created!");
+}
+/*........................................*/
+/*........................................*/
+void MyClass1__Method1(struct MyClass1* self, struct MyClass* param1)
+{
+    self->param = (*param1);
+}
+/*........................................*/
+/*........................................*/
+struct MyClass MyClass1__Method2(struct MyClass1* self)
+{
+    struct MyClass Out_0001;
+    MyClass__init__(&Out_0001, INT64_C(1));
+    return Out_0001;
+}
+/*........................................*/
+/*........................................*/
+void MyClass1__del__(struct MyClass1* self)
+{
+    if (!self->is_freed)
+    {
+        // pass
+        MyClass__del__(&self->param);
+        self->is_freed = 1;
+    }
+}
+/*........................................*/
+```
+
+### - C Program File Equivalent
+```C
+int main()
+{
+    struct MyClass1 obj;
+    struct MyClass Dummy_0000;
+    MyClass1__init__(&obj);
+    Dummy_0000 = MyClass1__Method2(&obj);
+    MyClass1__Method1(&obj, &Dummy_0000);
+    printf("%"PRId64"\n", obj.param.param1);
+    MyClass1__del__(&obj);
+    return 0;
+}
+```
+
+### - C _OUTPUT_
+
+```Shell
+MyClass1 Object created!
+MyClass Object created!
+1
+```
+
+## Limitations
+
+It's important to note that Pyccel does not support class wrappers, inheritance, properties, class variables, or inline class methods. The focus of Pyccel is primarily on core class functionality and memory management.
```

Comment:
Have you tested inline class methods? I plan to use this soon if it doesn't work then I'll need to fix it first :sweat_smile:

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1580/commits/3e0def115492fb89aae4d30826365da820416f87.
I need also to add tests in https://github.com/pyccel/pyccel/pull/1576.